src/qthotkey/gui.cc

```diff
@@ -0,0 +1,329 @@
+/*
+ *  This file is part of audacious-hotkey plugin for audacious
+ *
+ *  Copyright (C) 2020 i.Dark_Templar <darktemplar@dark-templar-archives.net>
+ *  Copyright (c) 2007 - 2008  Sascha Hlusiak <contact@saschahlusiak.de>
+ *  Name: gui.c
+ *  Description: gui.c
+ *
+ *  Part of this code is from itouch-ctrl plugin.
+ *  Authors of itouch-ctrl are listed below:
+ *
+ *  Copyright (c) 2006 - 2007 Vladimir Paskov <vlado.paskov@gmail.com>
+ *
+ *  Part of this code are from xmms-itouch plugin.
+ *  Authors of xmms-itouch are listed below:
+ *
+ *  Copyright (C) 2000-2002 Ville Syrjälä <syrjala@sci.fi>
+ *                         Bryn Davies <curious@ihug.com.au>
+ *                         Jonathan A. Davis <davis@jdhouse.org>
+ *                         Jeremy Tan <nsx@nsx.homeip.net>
+ *
+ *  audacious-hotkey is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  audacious-hotkey is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with audacious-hotkey; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include "plugin.h"
+#include "gui.h"
+
+#include <libaudcore/i18n.h>
+#include <libaudcore/preferences.h>
+
+#include <QtCore/QStringList>
+#include <QtGui/QKeyEvent>
+#include <QtGui/QMouseEvent>
+#include <QtWidgets/QApplication>
+#include <QtWidgets/QStyle>
+#include <QtX11Extras/QX11Info>
+
+#include <X11/XKBlib.h>
+
+static void destroy_callback();
+static void ok_callback();
+
+static QPrefWidget *last_instance = nullptr;
+
+static const char *event_desc[EVENT_MAX] = {
+	[EVENT_PREV_TRACK] = N_("Previous track"),
+	[EVENT_PLAY] = N_("Play"),
+	[EVENT_PAUSE] = N_("Pause/Resume"),
+	[EVENT_STOP] = N_("Stop"),
+	[EVENT_NEXT_TRACK] = N_("Next track"),
+	[EVENT_FORWARD] = N_("Step forward"),
+	[EVENT_BACKWARD] = N_("Step backward"),
+	[EVENT_MUTE] = N_("Mute"),
+	[EVENT_VOL_UP] = N_("Volume up"),
+	[EVENT_VOL_DOWN] = N_("Volume down"),
+	[EVENT_JUMP_TO_FILE] = N_("Jump to file"),
+	[EVENT_TOGGLE_WIN] = N_("Toggle player window(s)"),
+	[EVENT_SHOW_AOSD] = N_("Show On-Screen-Display"),
+	[EVENT_TOGGLE_REPEAT] = N_("Toggle repeat"),
+	[EVENT_TOGGLE_SHUFFLE] = N_("Toggle shuffle"),
+	[EVENT_TOGGLE_STOP] = N_("Toggle stop after current"),
+	[EVENT_RAISE] = N_("Raise player window(s)")
+};
+
+class QLineKeyEdit: public QLineEdit
```

Comment:
Class names should not be prefixed with Q, as they are not part of Qt.

Reply:
Thanks, fixed.